File: CMS/src/app/shared/PlanType.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { PlanType } from 'src/app/shared/PlanType.model';
import { BaseService } from './Base.service';
@Injectable({
  providedIn: 'root'
})
export class PlanTypeService extends BaseService {
  constructor(public httpClient: HttpClient) {
    super(httpClient);
    this.Controller = "PlanType";
  }  
}

File: Service/Implement/PlanTypeService.cs
namespace Service.Implement
{
    public class PlanTypeService : BaseService<PlanType, IPlanTypeRepository>
    , IPlanTypeService
    {
    private readonly IPlanTypeRepository _PlanTypeRepository;
    public PlanTypeService(IPlanTypeRepository PlanTypeRepository) : base(PlanTypeRepository)
    {
    _PlanTypeRepository = PlanTypeRepository;
    }
    }
    }

